neurom/core/population.py

```diff
@@ -27,42 +27,92 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 """Neuron Population Classes and Functions."""
+import logging
 
-from itertools import chain
+from morphio import MorphioError
+import neurom
+from neurom.exceptions import NeuroMError
+
+
+L = logging.getLogger(__name__)
 
 
 class Population:
     """Neuron Population Class.
 
-    Features:
-        - flattened collection of neurites.
-        - collection of somas, neurons.
-        - iterable-like iteration over neurons.
+    Offers an iterator over neurons within population, neurites of neurons, somas of neurons.
+    It does not store the loaded neuron in memory unless the neuron has been already passed
+    as loaded (instance of ``Neuron``).
     """
-    def __init__(self, neurons, name='Population'):
+    def __init__(self, files, name='Population', ignored_exceptions=(), cache=False):
         """Construct a neuron population.
 
         Arguments:
-            neurons: iterable of neuron objects.
-            name: Optional name for this Population.
+            files (collections.abc.Sequence[str|Path|Neuron]): collection of neuron files or
+                paths to them
+            name (str): Optional name for this Population
+            ignored_exceptions (tuple): NeuroM and MorphIO exceptions that you want to ignore when
+                loading neurons.
+            cache (bool): whether to cache the loaded neurons in memory. If false then a neuron
+                will be loaded everytime it is accessed within the population. Which is good when
+                population is big. If true then all neurons will be loaded upon the construction
+                and kept in memory.
         """
-        self.neurons = tuple(neurons)
-        self.somata = tuple(neu.soma for neu in neurons)
-        self.neurites = tuple(chain.from_iterable(neu.neurites for neu in neurons))
+        self._ignored_exceptions = ignored_exceptions
         self.name = name
+        if cache:
+            self._files = []
+            for f in files:
+                self._files.append(self._load_file(f))
```

Comment:
Is there a particular reason not to use list comprehension here?

Reply:
thanks, used it